cou/steps/plan.py

```diff
@@ -237,6 +243,37 @@ def _verify_highest_release_achieved(analysis_result: Analysis) -> None:
         )
 
 
+def _verify_nova_cloud_controller_scheduler_default_filters(
+    args: CLIargs, analysis_result: Analysis
+) -> None:
+    """Verify scheduler_default_filters option is set correctly for each OpenStack release.
+
+    :param args: CLI arguments
+    :type args: CLIargs
+    :param analysis_result: Analysis result.
+    :type analysis_result: Analysis
+    """
+    if args.upgrade_group != CONTROL_PLANE:
+        return
+
+    curr_release = analysis_result.current_cloud_o7k_release
+    nova_cloud_controllers = get_applications_by_charm_name(
+        analysis_result.apps_control_plane, "nova-cloud-controller"
+    )
+    for nova_cloud_controller in nova_cloud_controllers:
+        config = nova_cloud_controller.config.get("scheduler-default-filters", "")
+        if curr_release == "antelope" and "AvailabilityZoneFilter" in config:
```

Comment:
**issue:** 

`AvailabilityZoneFilter` may not be the only case.

I suggest to check the details of release notes of nova and understand all the available filter for different versions.

You can take a reference from the implementation of charm nova-cloud-controller: https://opendev.org/openstack/charm-nova-cloud-controller/src/commit/b915f8e6f81df6a776fc1ad6725207a199440a3b/hooks/nova_cc_context.py#L350-L396

Reply:
Can we leave other cases as out of scope for this PR? We can open a new issue to hunt for other cases.

Reply:
I agree with Samuel, it's too early to check all cases from changelogs. This is the one we encountered, and mentioned explicitly in the [changelog](https://docs.openstack.org/charm-guide/latest/release-notes/2023.2-bobcat.html#nova-availabilityzonefilter-removal-in-2023-2).

Reply:
@jneo8 please consider creating new issue to track other potential issues that have not been observed (e.g. from ussuri -> victorial, it's possible becuase `RetryFilter` is not in victorial)

Reply:
I created https://github.com/canonical/charmed-openstack-upgrader/issues/600 to track this. :)